api/scpca_portal/models/dataset.py

```diff
@@ -67,6 +79,106 @@ class Meta:
     def __str__(self):
         return f"Dataset {self.id}"
 
+    @classmethod
+    def get_ccdl_dataset(cls, ccdl_name, project_id: str | None = None):
+        dataset = cls(is_ccdl=True, ccdl_name=ccdl_name, ccdl_project_id=project_id)
+        dataset.format = dataset.ccdl_type["format"]
+        dataset.data = dataset.get_ccdl_data()
+        return dataset
+
+    def get_ccdl_data(self) -> Dict:
+        projects = Project.objects.all()
+        if self.ccdl_project_id:
+            projects = projects.filter(scpca_id=self.ccdl_project_id)
+
+        data = {}
+        for project in projects:
+            samples = project.samples
+            if self.ccdl_type.get("excludes_multiplexed"):
+                samples = samples.filter(has_multiplexed_data=False)
+
+            data[project.scpca_id] = {
+                "merge_single_cell": self.ccdl_type.get("includes_merged"),
+                "includes_bulk": True,
+                Modalities.SINGLE_CELL: [],
+                Modalities.SPATIAL: [],
+            }
+
+            # Data dataset types
+            if modality := self.ccdl_type.get("modality"):
+                data[project.scpca_id][modality].extend(
+                    samples.filter(modality=modality).values_list("scpca_id", flat=True)
+                )
+            # Metadata only dataset types
+            else:
+                data[project.scpca_id][Modalities.SINGLE_CELL] = samples.filter(
+                    modality=Modalities.SINGLE_CELL
+                ).values_list("scpca_id", flat=True)
+                data[project.scpca_id][Modalities.SPATIAL] = samples.filter(
+                    modality=Modalities.SPATIAL
+                ).values_list("scpca_id", flat=True)
+
+        return data
+
+    def validate_samples(self) -> bool:
```

Comment:
- remove logs
- return tuple of: (success, error_message)

Reply:
alternatively just cut this out along with my comment above and lets add this in when we need it